frontend/packages/gitops-plugin/src/components/list/GitOpsList.tsx

```diff
@@ -1,46 +1,119 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
-import GitOpsListItem from './GitOpsListItem';
-import { Stack, StackItem, Split, SplitItem } from '@patternfly/react-core';
-import { K8sResourceKind } from '@console/internal/module/k8s';
 import { GitOpsAppGroupData } from '../utils/gitops-types';
 import GitOpsEmptyState from '../GitOpsEmptyState';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
+import { Button } from '@patternfly/react-core';
+import { Table, TableRow, TableData, RowFunction } from '@console/internal/components/factory';
+import { history, ExternalLink } from '@console/internal/components/utils';
+import { routeDecoratorIcon } from '@console/dev-console/src/components/import/render-utils';
 import './GitOpsList.scss';
 
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-4', 'col-sm-4', 'col-xs-6'),
+  classNames('col-lg-3', 'col-md-4', 'col-sm-4', 'col-xs-6'),
+  classNames('col-lg-2', 'col-md-2', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-2', 'col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+];
 interface GitOpsListProps {
-  appGroups: GitOpsAppGroupData[];
+  appGroups: GitOpsAppGroupData;
   emptyStateMsg: string;
-  argocdLink?: K8sResourceKind;
+  props?: any;
 }
 
-const GitOpsList: React.FC<GitOpsListProps> = ({ appGroups, emptyStateMsg, argocdLink }) => {
+const GitOpsList: React.FC<GitOpsListProps> = ({ appGroups, emptyStateMsg, props }) => {
   const { t } = useTranslation();
+
+  const handleClick = (appGroup) => {
+    history.push(`/environments/${appGroup.name}?url=${appGroup.repo_url}`);
+  };
+
+  const GitOpsTableHeader = () => {
```

Comment:
Would be better to pull this out into its own file. Keeping this in the render context of component is not a good idea.